peer.go

```diff
@@ -433,6 +433,12 @@ func (p *Peer) getConnectionRelay(timeout time.Duration) (*Connection, error) {
 		return activeConn, nil
 	}
 
+	// Use the lower timeout value of the call timeout and the relay connection timeout.
+	timeout := callTimeout
+	if timeout > relayMaxConnTimeout && relayMaxConnTimeout > 0 {
+		timeout = relayMaxConnTimeout
+	}
```

Comment:
This looks like it gets plumbed through by `NewContextBuilder` as the context's timeout, but `NewContextBuilder` seems to support a `ConnectTimeout`, which is used by `Connect`.

How do those differ for this situation? Don't we want to use `ConnectTimeout` for this case?

Reply:
Good digging, the `ConnectTimeout` is intended for calls where you want to limit how long is spent connecting. E.g., I may have a call that has a 5 second timeout, but I only want 1 second spent connecting (so there's enough time to try on a different peer).

The `Connect` logic prefers to use the `ConnectTimeout` is specified, but otherwise, falls back to the context timeout:
https://github.com/uber/tchannel-go/blob/dev/channel.go#L563

In our case, since we're creating the context purely for the connection timeout, there's no need to specify the `ConnectTimeout` (since it'll fallback to the context timeout).

Reply:
I think this makes sense - this makes it so that the relay connection fails faster with `timeout` than it normally would?

Reply:
The `ConnectTimeout` is useful for standard non-relay clients (e.g., `client.CallMethod(ctx, ...)` where the method may need more time separate to connects). If you don't specify it, the `ctx` timeout is used. In this case, setting `ConnectTimeout` wouldn't have any impact, since the only purpose of the `ctx` created below is for the connect timeout.

The `ConnectTimeout` is an advanced option that makes sense for normal clients, but in the relay code path, when we create a context purely for connections, it doesn't make sense to use.